### PR TITLE
doc: Add document of RT performance tuning.

### PR DESCRIPTION
**Pull request by lirui34**
Add document of RT performance tuning.

Signed-off-by: lirui34 <ruix.li@intel.com>

### CODE REVIEW

File: doc/tutorials/realtime_performance_tuning.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+Trace and Data Collection for ACRN Real-Time(RT) Performance Tuning
+###################################################################
+
+VM exits analysis for ACRN RT performance
+*****************************************
+
+VM exits in response to certain instructions and events are a key source of 
+performance degradation in virtual machines. During the runtime of hard RTVM 
+of ACRN, there are still some instructions and events which will impact the 
+RT latency's determinacy.
```
Comment (lyan3) [inline doc/tutorials/realtime_performance_tuning.rst]: "determinism" is better.

File: doc/tutorials/realtime_performance_tuning.rst [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+         clock_gettime(&now)
+         latency = calcdiff(now, next)
+         …
+         next += interval
+   }
+
+Time point ``now`` is the actual point at which the cyclictest is wakeuped and 
+scheduled. Time point ``next`` is the expected point at which we want the cyclictest 
+to be wakeuped and scheduled. Here we can get the latency by now - next. We don't 
+want to see VM exits during ``next`` through ``now``. So define the start point of
```
Comment (lyan3) [inline doc/tutorials/realtime_performance_tuning.rst]: wakeuped -> woken up

File: doc/tutorials/realtime_performance_tuning.rst [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+#. Add timestamps (in TSC) at ``next`` and ``now``.
+#. Capture the log with the above timestamps in RTVM.
+#. Capture the acrntrace log in Service VM at the same time.
+
+Offline analysis
+================
+
+#. Convert the raw trace data to human readable format.
+#. Merge the logs in RTVM and ACRN hypervisor trace based on timestamps (in TSC).
+#. Check if there are VM exits which hit the critical section, the pattern is as following:
```
Comment (lyan3) [inline doc/tutorials/realtime_performance_tuning.rst]: if there is any VM exit within the critical sections

Comment (lyan3) [inline doc/tutorials/realtime_performance_tuning.rst]: the pattern is as follows:

File: doc/tutorials/realtime_performance_tuning.rst (hunk 1 above)
Comment (lirui34) [reply doc/tutorials/realtime_performance_tuning.rst]: fixed.

File: doc/tutorials/realtime_performance_tuning.rst (hunk 2 above)
Comment (lirui34) [reply doc/tutorials/realtime_performance_tuning.rst]: fixed.

File: doc/tutorials/realtime_performance_tuning.rst (hunk 3 above)
Comment (lirui34) [reply doc/tutorials/realtime_performance_tuning.rst]: fixed.

Comment (lirui34) [reply doc/tutorials/realtime_performance_tuning.rst]: fixed.

File: doc/tutorials/realtime_performance_tuning.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+         clock_nanosleep(&next)
+         clock_gettime(&now)
+         latency = calcdiff(now, next)
+         …
+         next += interval
+   }
+
+Time point ``now`` is the actual point at which the cyclictest is wakeuped and 
+scheduled. Time point ``next`` is the expected point at which we want the cyclictest 
+to be woken up and scheduled. Here we can get the latency by now - next. We don't
```
Comment (KaigeFu) [inline doc/tutorials/realtime_performance_tuning.rst]: Better to change now - next to ```now - next`

Comment (lirui34) [reply doc/tutorials/realtime_performance_tuning.rst]: Fixed :smile:

**Pull request merged**